src/core/debt_register.sol

```diff
@@ -105,71 +107,70 @@ contract Beans is DSNote {
         z = x / y;
     }
 
-    function incLoanDebt(uint loan, uint fee, uint wad) public auth note {
-        require(now == fees[fee].rho);
-        pie[loan] = add(pie[loan], calcPie(fees[fee].chi, wad));
-        incTotalDebt(fee, wad);
+    function incLoanDebt(uint loan, uint rate, uint wad) public auth note {
+        require(now == rates[rate].lastUpdated);
+        debtBalance[loan] = add(debtBalance[loan], calcDebtBalance(rates[rate].rateIndex, wad));
+        incTotalDebt(rate, wad);
     }
 
-    function decLoanDebt(uint loan, uint fee, uint wad) public auth note {
-        require(now == fees[fee].rho);
-        pie[loan] = sub(pie[loan], calcPie(fees[fee].chi, wad));
-        decTotalDebt(fee, wad);
+    function decLoanDebt(uint loan, uint rate, uint wad) public auth note {
+        require(now == rates[rate].lastUpdated);
+        debtBalance[loan] = sub(debtBalance[loan], calcDebtBalance(rates[rate].rateIndex, wad));
+        decTotalDebt(rate, wad);
     }
 
-    function compounding(uint fee) public view returns (uint, uint, uint) {
-        uint48 rho = fees[fee].rho;
-        require(now >= rho);
-        uint speed = fees[fee].speed;
+    function compounding(uint rate) public view returns (uint, uint, uint) {
+        uint48 lastUpdated = rates[rate].lastUpdated;
+        require(now >= lastUpdated);
+        uint ratePerSecond = rates[rate].ratePerSecond;
 
-        uint chi = fees[fee].chi;
-        uint debt_ = fees[fee].debt;
+        uint rateIndex = rates[rate].rateIndex;
+        uint debt_ = rates[rate].debt;
 
         // compounding in seconds
-        uint latest = rmul(rpow(speed, now - rho, ONE), chi);
-        uint chi_ = rdiv(latest, chi);
-        uint wad = rmul(debt_, chi_) - debt_;
-        return (latest, chi_, wad);
-    }
-
-    // --- Fee Accumulation ---
-    function drip(uint fee) public {
-        if (now >= fees[fee].rho) {
-            (uint latest, , uint wad) = compounding(fee);
-            fees[fee].chi = latest;
-            fees[fee].rho = uint48(now);
-            incTotalDebt(fee, wad);   
+        uint latestRateIndex = rmul(rpow(ratePerSecond, now - lastUpdated, ONE), rateIndex);
+        uint rateIndex_ = rdiv(latestRateIndex, rateIndex);
+        uint wad = rmul(debt_, rateIndex_) - debt_;
+        return (latestRateIndex, rateIndex_, wad);
+    }
+
+    // --- Rate Accumulation ---
+    function drip(uint rate) public {
+        if (now >= rates[rate].lastUpdated) {
+            (uint latestRateIndex, , uint wad) = compounding(rate);
+            rates[rate].rateIndex = latestRateIndex;
+            rates[rate].lastUpdated = uint48(now);
+            incTotalDebt(rate, wad);   
         }
     }
 
-    function burden(uint loan, uint fee) public view returns (uint) {
-        uint chi = fees[fee].chi;
-        if (now >= fees[fee].rho) {
-            (chi, ,) = compounding(fee);
+    function getCurrentDebt(uint loan, uint rate) public view returns (uint) {
```

Comment:
we don't actually use this function anywhere

Reply:
We use it in the UI :)

Reply:
🙈